### ISSUE TITLE
Warning Check on Unused Parameter hides Error Check about Missing Parameter Type

### ISSUE THREAD

**Issue by phene** [aws-cloudformation/cfn-python-lint#1223]
*cfn-lint version: cfn-lint 0.25.3*

Parameters defined in a template, but not directly used, are not validated for missing attributes like `Type`.

For various reasons, we want to include parameters in our templates that are not used by resources in the templates and therefore disable `W2001` When this happens, the following template will not fail cfn-lint. If I uncomment the `Metadata` section, I will finally see the `E1012` failure. I should not have to resolve a Warning in order to unmask an Error.

```yaml
Parameters:
  Foo:
    Description: "Foo?"
Conditions:
  AlwaysFalse: !Equals [ true, false ]
Resources:
  # Metadata:
  #   Foo: !Ref Foo
  NullResource:
    Type: Custom::NullResource
    Condition: AlwaysFalse
```
